cpp/src/io/parquet/bloom_filter_reader.cu

```diff
@@ -609,10 +615,16 @@ std::optional<std::vector<std::vector<size_type>>> aggregate_reader_metadata::ap
   // Return early if no column with equality predicate(s)
   if (equality_col_schemas.empty()) { return std::nullopt; }
 
+  using policy_type = cuco::arrow_filter_policy<cuda::std::byte, cudf::hashing::detail::XXHash_64>;
+  size_t constexpr alignment =
+    policy_type::words_per_block * sizeof(typename policy_type::word_type);
+  auto aligned_mr =
+    rmm::mr::aligned_resource_adaptor(cudf::get_current_device_resource(), alignment);
```

Comment:
Could we create the aligned `mr` within `read_bloom_filters` instead of passing it as a parameter? Typically, passing an `mr` in libcudf implies that the return value is allocated using the given `mr`. However, since this function returns void, passing an `mr` might be misleading.

Reply:
Right, I created the `mr` here as it will be used by `read_bloom_filters` to allocate and return the buffers back, the same buffers will be used to do bloom filter stuff and dealloc along with this `mr` when we return.

Reply:
@bdice could you also confirm if the use of `aligned_mr` is correct or can it be moved inside `read_bloom_filter_data` as well?

Reply:
The `aligned_mr` needs to be alive at least as long as its allocations are in use. It wraps an upstream memory resource, and then internally tracks a map of unaligned pointers from the upstream to aligned pointers that it hands back. https://github.com/rapidsai/rmm/blob/d750a6e8578a4e61159faae59baeb4470e2ed81f/include/rmm/mr/device/aligned_resource_adaptor.hpp#L209

If you ask the upstream memory resource to deallocate the aligned pointers, it won't know how to do the right thing. Thus, for lifetime reasons, I think we need `aligned_mr` to be passed into the functions, as the `aligned_mr` must outlive the `bloom_filter_data` that is returned.

Reply:
I'm having a bit of trouble tracing the call stack in the diff here. Whenever the aligned data is no longer needed, it is safe to discard the `aligned_mr`.

Reply:
> Whenever the aligned data is no longer needed, it is safe to discard the `aligned_mr`. Functions returning aligned data need to take the `aligned_mr` as a parameter.

I believe this is well covered. Thanks for the clarification.